tests/test_app.py

```diff
@@ -361,6 +370,42 @@ def test_node_send_stats_with_unknown_node(self):
         self.assertEqual(400, re.status_code)
         self.assertEqual({'error': 'node key not found'}, re.json)
 
+    def test_restrict_by_ip_fail(self):
+        settings.RESTRICT_BY_IP_ENABLED = True
+        settings.ALLOWED_IP_ADDRESSES = ''
```

Comment:
I would add at least single ip address here to make sure that ips are compared - current tests can pass by implementing alternative simpler decorator implementation:
```py
if settings.ALLOWED_IP_ADDRESSES != '':
  # raise 403
```
You don't actually have to compare ips to pass these tests :)